roles/openshift_common/tasks/main.yml

```diff
@@ -34,6 +34,26 @@
       use_manageiq: "{{ openshift_use_manageiq | default(None) }}"
       data_dir: "{{ openshift_data_dir | default(None) }}"
 
+- name: Gather current version of docker
+  command: >
+    rpm -q docker --queryformat '%{version}'
+  register: installed_docker
+  failed_when: false
+  changed_when: false
+  
+- name: Gather latest version of common package
+  shell: >
+    yum list available -e 0 -q "{{ openshift.common.service_type}}" 2>&1 | tail -n +2 | awk '{ print $2 }' | sort -r | tr '\n' ' ' | tail -n 1 | cut -d '.' -f1,2
+  register: latest_package
+  failed_when: false
+  changed_when: false
+
+- name: Downgrade docker to 1.8.2 if we're installing a version older than 3.2
+  command: >
+    {{ ansible_pkg_mgr }} -y downgrade docker-1.8.2
+  notify: restart docker
+  when: " '1.9' in installed_docker.stdout and latest_package.stdout in ['3.0','3.1'] "
+
```

Comment:
The problem I ran into is that openshift_common requires the base package and that comes before we bother with docker role, which is fine if docker hasn't been installed, however if it has been installed already and it's version 1.9 then the base package will fail to install if it's < 3.2.

If openshift_common were to depend on openshift_docker I could teach the docker role to downgrade if necessary.